### DISCUSSION TITLE
Capture a picture while mediamtx is running

### DISCUSSION THREAD

**Discussion by boos87** [bluenviron/mediamtx#3219] [Questions]
### Question

How can i capture pictures in raw directly from the camera, while running mediamtx is running?

For my project, i had setup a RPI Zero 2 with mediamtx. The stream is implement in a website on a webserver is running on a Raspberry Pi 5 (at least 8 streams necessary from 8 RPI Zero 2) to see a preview of the camera(s). While running the mediamtx i can´t capture a picture from the camera.

```
$ libcamera-raw -o 20240409_test2.jpg
[0:01:23.816516521] [643]  INFO Camera camera_manager.cpp:297 libcamera v0.0.5+83-bde9b04f
[0:01:23.892325928] [644]  INFO RPI vc4.cpp:437 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media2 and ISP device /dev/media0
[0:01:23.892511077] [644]  INFO RPI pipeline_base.cpp:1101 Using configuration file '/usr/share/libcamera/pipeline/rpi/vc4/rpi_apps.yaml'
[0:01:23.895136141] [643]  INFO Camera camera.cpp:840 Pipeline handler in use by another process
ERROR: *** failed to acquire camera /base/soc/i2c0mux/i2c@1/ov5647@36 ***
```

Isn´t it possible to capture pictures, while running mediamtx? What can i do? I want to capture pirctures every 500ms and after that, a little bit image processing.

**Comment by aler9** [comment]
Hello, since the camera feed is being streamed into the server, you have to read that feed from the server and not from the camera.

In order to extract images for preview purposes or processing purposes, you can use `ffmpeg` inside `runOnReady`:

```yml
paths:
  mycamera:
    source: rpicamera
    runOnReady: ffmpeg -i rtsp://localhost:8554/mycamera -vf fps=1 out%d.jpg
    runOnReadyRestart: yes
```

this will generate an image every second, you can tune the `fps` parameter in order to decrease or increase the interval.

[a comment by an automated account is not shown]
